instrumentation-api-caching/src/main/java/io/opentelemetry/instrumentation/api/caching/Caffeine2Cache.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.caching;
+
+import io.opentelemetry.instrumentation.api.internal.shaded.caffeine2.cache.Caffeine;
```

Comment:
I was hoping it's possible to use the same package but put caffeine3 into the mrjar folder. And with just one implementation jvm would automatically pick the correct one. Could it work?

Reply:
Unfortunately afaik boot loader does not support mrjars.